src/main/java/io/codearte/jfairy/Bootstrap.java

```diff
@@ -32,22 +35,32 @@
 
 	private static final String DATA_FILE_PREFIX = "jfairy";
 
-	public static Fairy createFairy(Locale locale, String filePrefix, Random random) {
 
-		FairyModule fairyModule = getFairyModuleForLocale(locale, random);
+	public static Fairy createFairy(Optional<DataMaster> dataMaster, Locale locale, String filePrefix, Random random) {
+
+		FairyModule fairyModule = getFairyModuleForLocale(dataMaster, locale, random);
 
 		Injector injector = Guice.createInjector(fairyModule);
 
-		DataMaster dataMaster = injector.getInstance(DataMaster.class);
+		FairyFactory instance = injector.getInstance(FairyFactory.class);
+
+		DataMaster createdDataMaster = injector.getInstance(DataMaster.class);
```

Comment:
No, we don't, that's why the conditional is there. The default DataMaster is created if no DataMaster is passed as argument of createFairy() method. However, I have just realised that I should add yet another implementation of this method, with both custom Locale and DataMaster. Will add it today.